src/components/QualityRatingButton.js

```diff
@@ -13,10 +13,10 @@ const RatingButton = styled.button`
   background: none;
 `;
 
-export default function QualityRatingButton({ active, className }) {
+export default function QualityRatingButton({ active, className, onClick }) {
   return (
     <RatingButton active={active} className={className}>
-      <Favorite />
+      <Favorite onClick={onClick} />
```

Comment:
Set the `onClick` to the `RatingButton, to make sure, that the onClick is fired on Button click and not on Icon click. 
There might be no visual difference, but if you add a `padding` to the button, you would face some weired user experience.